### PR TITLE
fix #33 Project update to 2023.02.xx

### LINKED ISSUE #33: Project Update
It is requested to update the downstream project to the latest 2023.02.xx.

More to be done within this issue:
- [x] Create a production branch from the current master before starting the update work 
- [x] Create a tag release 1.0.0 from the current production branch
- [x] Perform the requested update on the master branch

### PR DESCRIPTION
**Pull request by MV88**
fix #33 Project update to 2023.02.xx

### CODE REVIEW

File: export.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export LDAP_URL="ldap://this.will.not.work:389/dc=xxxxx,dc=xxxxx,dc=xxxxx"
```
Comment (offtherailz) [inline export.sh:1]: I think this is a script for testing. Document it or drop it.

File: web/src/main/resources/mapstore.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+extensions.registry=extensions.json
```
Comment (offtherailz) [inline web/src/main/resources/mapstore.properties:1]: Is this really needed for a project? I don't remember this ... Where did you find this change to apply?

File: web/src/main/resources/sample_categories.xml [hunk 3]
```diff
@@ -18,4 +18,13 @@
...
     </Category>
+    <Category>
+        <name>TEMPLATE</name>
+    </Category>
+    <Category>
+        <name>USERSESSION</name>
+    </Category>
+    <Category>
+        <name>PERMALINK</name>
+    </Category>
```
Comment (offtherailz) [inline web/src/main/resources/sample_categories.xml:29]: These will need to be created if the database already exists.

File: web/src/main/webapp/WEB-INF/classes/geostore-spring-security.xml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                <constructor-arg>
+                        <bean
+                                class="org.springframework.security.ldap.authentication.BindAuthenticator">
+                                <constructor-arg ref="contextSource" />
+
+                                <property name="userSearch">
+                                        <bean id="userSearch"
+                                                class="org.springframework.security.ldap.search.FilterBasedLdapUserSearch">
+                                                <constructor-arg index="0" value="" />
+                                                <constructor-arg index="1" value="(&amp;(objectCategory=${LDAP_USER_SEARCH_CATEGORY})(|#LDAP_GROUPS)(sAMAccountName={0}))" />
```
Comment (offtherailz) [inline web/src/main/webapp/WEB-INF/classes/geostore-spring-security.xml:125]: has this been already tested ?

**Review by offtherailz — CHANGES_REQUESTED** [review]
I think it we are almost there. 
- [x] See comments inline.
- [x] Please document on the readme how to run locally for future tests, development and update, I was not able to do it running standard commands. 
- [x] Please document also LDAP custom variables, if any. 
- [x] package-lock.json is old. What if we commit an update for it?
- [ ] test properly
- [ ] Make sure to apply upgrades also on installation (compile a list, if I missed something), see details below.

### Documentation notes

Readme file I think is enough to  document how to run locally and how to configure (when not standard LDAP configuration, in case link the onine documentation).

**also** update node version indication on readme.

### Testing notes

Make sure to have tested this project widely (I'll not do, and I wasn't able to do it because of reasons above).
Especially things that had to be ported manually during upgrade, like:
- embedded templates for all content to be working
- ldap configuration
- printing

### Deploy notes

During installation, take also into account to:
- create the missing categories on installation (PERMALINK, USERSESSION, TEMPLATE)
- database schema probably changed. meke sure to update `geostore-datasource-ovr.properties` to automatically apply changes, or to apply the proper changes to the schema manually to the existing geostore.
```
# Automatic create-update database mode
geostoreEntityManagerFactory.jpaPropertyMap[hibernate.hbm2ddl.auto]=update

File: web/src/main/resources/sample_categories.xml (hunk 3 above)
Comment (MV88) [reply web/src/main/resources/sample_categories.xml:29]: yes, it's a devop task

File: web/src/main/webapp/WEB-INF/classes/geostore-spring-security.xml (hunk 4 above)
Comment (MV88) [reply web/src/main/webapp/WEB-INF/classes/geostore-spring-security.xml:125]: this was the original implementation  and i suppose it was tested back in the days. we will test it after deploy  update i guess

File: export.sh (hunk 1 above)
Comment (MV88) [reply export.sh:1]: i'll document it and i'll include in start commands

**Review by offtherailz — CHANGES_REQUESTED** [review]
I updated the README with some testing hints. They are not optimal but at least we can try it with a mock LDAP.  For the future we can improve them.

- [x] There are a few differences in theme. Take a look if you can uniform them with the standard theme. I don't see in fact a lot of customization to be maintained. Isn't it?
image
- [ ] Need to document the properties. I asked @randomorder to provide a sample (without credentials) of the variables to give an Idea of the final result. 
- [ ] I still see `mapstore.properties` set as I said here, but not answer here. https://github.com/geosolutions-it/MapStore-C179/pull/34#discussion_r1407436008
- [ ] How can we make sure that the changes will be applied on deploy (*Deploy notes* [here](https://github.com/geosolutions-it/MapStore-C179/pull/34#pullrequestreview-1752367895) ). Should we write an update note @randomorder ?

**Comment by MV88** [comment]
>  There are a few differences in theme. Take a look if you can uniform them with the standard theme. I don't see in fact a lot of customization to be maintained. Isn't it?

it appears that some customized variables in the old version of the project were not used or the theme structure was different, 
but now i have aligned these 
@ms2-color-disabled: #5C9FB4;
@ms2-color-text-disabled: #ffffff;
to become 
@ms-disabled-bg: #eee;
@ms-disabled-color: #999;

the fix will arrive in a new commit

> I still see mapstore.properties set as I said here, but not answer here. https://github.com/geosolutions-it/MapStore-C179/pull/34#discussion_r1407436008

I have just aligned the properties file as listed in the checklist of migration guidelines, do i have to avoid to do this change?

**Review by offtherailz — CHANGES_REQUESTED** [review]
waiting for extension.json to be checked.
Rest of the notes have to be passed to @randomorder

**Comment by tdipisa** [comment]
> waiting for extension.json to be checked. Rest of the notes have to be passed to @randomorder

@MV88 after merging this PR a new issues for the deployment should be created for @randomorder by reporting all relevant activities to take into account point by point.

**Review by offtherailz — APPROVED** [review]

**Review by tdipisa — CHANGES_REQUESTED** [review]
@MV88 you may want to check compile failures before merging this PR.

**Comment by tdipisa** [comment]
> > waiting for extension.json to be checked. Rest of the notes have to be passed to @randomorder
> 
> @MV88 after merging this PR a new issues for the deployment should be created for @randomorder by reporting all relevant activities to take into account point by point.

That's the related issue @EddyCatt @randomorder 

https://github.com/geosolutions-it/MapStore-C179/issues/35

**Review by tdipisa — APPROVED** [review]

**Pull request merged**